// option-t/lib/Result.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.compatToPlainResult = exports.compatToClassicResult = exports.Err = exports.Ok = exports.createErr = exports.createOk = exports.ResultBase = void 0;
/**
 *  @deprecated
 *      We keep this file only for backward compatibility.
 *      See https://github.com/karen-irc/option-t/issues/459
 */
var ClassicResult_1 = require("./ClassResult/ClassicResult");
Object.defineProperty(exports, "ResultBase", { enumerable: true, get: function () { return ClassicResult_1.ClassicResultBase; } });
Object.defineProperty(exports, "createOk", { enumerable: true, get: function () { return ClassicResult_1.createClassicOk; } });
Object.defineProperty(exports, "createErr", { enumerable: true, get: function () { return ClassicResult_1.createClassicErr; } });
Object.defineProperty(exports, "Ok", { enumerable: true, get: function () { return ClassicResult_1.ClassicOkConstructor; } });
Object.defineProperty(exports, "Err", { enumerable: true, get: function () { return ClassicResult_1.ClassicErrConstructor; } });
var compat_1 = require("./ClassResult/compat");
Object.defineProperty(exports, "compatToClassicResult", { enumerable: true, get: function () { return compat_1.compatToClassicResult; } });
Object.defineProperty(exports, "compatToPlainResult", { enumerable: true, get: function () { return compat_1.compatToPlainResult; } });
