// option-t/cjs/Option.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.compatToPlainOption = exports.compatToClassicOption = exports.None = exports.Some = exports.createNone = exports.createSome = exports.OptionBase = void 0;
/**
 *  @deprecated
 *      We keep this file only for backward compatibility.
 *      See https://github.com/karen-irc/option-t/issues/459
 */
var ClassicOption_1 = require("./ClassOption/ClassicOption");
Object.defineProperty(exports, "OptionBase", { enumerable: true, get: function () { return ClassicOption_1.ClassicOptionBase; } });
Object.defineProperty(exports, "createSome", { enumerable: true, get: function () { return ClassicOption_1.createClassicSome; } });
Object.defineProperty(exports, "createNone", { enumerable: true, get: function () { return ClassicOption_1.createClassicNone; } });
Object.defineProperty(exports, "Some", { enumerable: true, get: function () { return ClassicOption_1.ClassicSomeConstructor; } });
Object.defineProperty(exports, "None", { enumerable: true, get: function () { return ClassicOption_1.ClassicNoneConstructor; } });
var compat_1 = require("./ClassOption/compat");
Object.defineProperty(exports, "compatToClassicOption", { enumerable: true, get: function () { return compat_1.compatToClassicOption; } });
Object.defineProperty(exports, "compatToPlainOption", { enumerable: true, get: function () { return compat_1.compatToPlainOption; } });
