// option-t/lib/Undefinable/index.mjs
export { isNotUndefined, isUndefined } from "./Undefinable.mjs"; // XXX: `and()` operation is equivalent of `a && b` so we don't ship it by default set.
//export { andForUndefinable as and} from './and';

export { andThenForUndefinable as andThen } from "./andThen.mjs";
export { andThenAsyncForUndefinable as andThenAsync } from "./andThenAsync.mjs";
export { expectNotUndefined as expect } from "./expect.mjs";
export { mapForUndefinable as map } from "./map.mjs";
export { mapAsyncForUndefinable as mapAsync } from "./mapAsync.mjs";
export { mapOrForUndefinable as mapOr } from "./mapOr.mjs";
export { mapOrAsyncForUndefinable as mapOrAsync } from "./mapOrAsync.mjs";
export { mapOrElseForUndefinable as mapOrElse } from "./mapOrElse.mjs";
export { mapOrElseAsyncForUndefinable as mapOrElseAsync } from "./mapOrElseAsync.mjs"; // XXX: `or()` operation is equivalent of `a || b` so we don't ship it by default set.
// export { orForUndefinable as or} from './or';

export { orElseForUndefinable as orElse } from "./orElse.mjs";
export { orElseAsyncForUndefinable as orElseAsync } from "./orElseAsync.mjs";
export { tapUndefinable as tap } from "./tap.mjs";
export { unwrapUndefinable as unwrap } from "./unwrap.mjs";
export { unwrapOrFromUndefinable as unwrapOr } from "./unwrapOr.mjs";
export { unwrapOrElseFromUndefinable as unwrapOrElse } from "./unwrapOrElse.mjs";
export { unwrapOrElseAsyncFromUndefinable as unwrapOrElseAsync } from "./unwrapOrElseAsync.mjs";
