// option-t/lib/Maybe/index.mjs
export { isNotNullAndUndefined, isNullOrUndefined } from "./Maybe.mjs"; // XXX: `and()` operation is equivalent of `a && b` so we don't ship it by default set.
//export { andForMaybe as and } from './and';

export { andThenForMaybe as andThen } from "./andThen.mjs";
export { andThenAsyncForMaybe as andThenAsync } from "./andThenAsync.mjs";
export { expectNotNullAndUndefined as expect } from "./expect.mjs";
export { mapForMaybe as map } from "./map.mjs";
export { mapAsyncForMaybe as mapAsync } from "./mapAsync.mjs";
export { mapOrForMaybe as mapOr } from "./mapOr.mjs";
export { mapOrAsyncForMaybe as mapOrAsync } from "./mapOrAsync.mjs";
export { mapOrElseForMaybe as mapOrElse } from "./mapOrElse.mjs";
export { mapOrElseAsyncForMaybe as mapOrElseAsync } from "./mapOrElseAsync.mjs"; // XXX: `or()` operation is equivalent of `a || b` so we don't ship it by default set.
// export { orForMaybe as or } from './or';

export { orElseForMaybe as orElse } from "./orElse.mjs";
export { orElseAsyncForMaybe as orElseAsync } from "./orElseAsync.mjs";
export { tapMaybe as tap } from "./tap.mjs";
export { unwrapMaybe as unwrap } from "./unwrap.mjs";
export { unwrapOrFromMaybe as unwrapOr } from "./unwrapOr.mjs";
export { unwrapOrElseFromMaybe as unwrapOrElse } from "./unwrapOrElse.mjs";
export { unwrapOrElseAsyncFromMaybe as unwrapOrElseAsync } from "./unwrapOrElseAsync.mjs";
